x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -0,0 +1,391 @@
+	defaultUsername   = "elastic"
+	defaultPassword   = "changeme"
+	defaultTokenName  = "Default"
+)
+
+func newContainerCommand(flags *globalFlags, _ []string, streams *cli.IOStreams) *cobra.Command {
+	return &cobra.Command{
+		Hidden: true, // not exposed over help; used by container entrypoint only
+		Use:    "container",
+		Short:  "Bootstrap Elastic Agent to run inside of a container",
+		Long: `This should only be used as an entrypoint for a container. This will prepare the Elastic Agent using
+environment variables to run inside of the container.
+
+The following actions are possible and grouped based on the actions.
+
+* Preparing Kibana for Fleet
+  This prepares the Fleet plugin that exists inside of Kibana. This must either be enabled here or done externally
+  before Fleet Server will actually successfully start.
+
+  KIBANA_FLEET_SETUP - set to 1 enables this setup
+  KIBANA_FLEET_HOST - kibana host to enable Fleet on [$KIBANA_HOST]
+  KIBANA_FLEET_USERNAME - kibana username to enable Fleet [$KIBANA_USERNAME]
+  KIBANA_FLEET_PASSWORD - kibana password to enable Fleet [$KIBANA_PASSWORD]
+
+* Bootstrapping Fleet Server
+  This bootstraps the Fleet Server to be run by this Elastic Agent. At least one Fleet Server is required in a Fleet
+  deployment for other Elastic Agent to bootstrap.
+
+  FLEET_SERVER_ENABLE - set to 1 enables bootstrapping of Fleet Server (forces FLEET_ENROLL enabled)
+  FLEET_SERVER_ELASTICSEARCH_HOST - elasticsearch host for Fleet Server to communicate with [$ELASTICSEARCH_HOST]
+  FLEET_SERVER_ELASTICSEARCH_USERNAME - elasticsearch username for Fleet Server [$ELASTICSEARCH_USERNAME]
+  FLEET_SERVER_ELASTICSEARCH_PASSWORD - elasticsearch password for Fleet Server [$ELASTICSEARCH_PASSWORD]
+  FLEET_SERVER_POLICY_NAME - name of policy for the Fleet Server to use for itself [$FLEET_TOKEN_POLICY_NAME]
+  FLEET_SERVER_POLICY_ID - policy ID for Fleet Server to use for itself ("Default Fleet Server policy" used when undefined)
+  FLEET_SERVER_HOST - binding host for Fleet Server HTTP (overrides the policy)
+  FLEET_SERVER_PORT - binding port for Fleet Server HTTP (overrides the policy)
+  FLEET_SERVER_CERT - path to certificate to use for HTTPS endpoint
+  FLEET_SERVER_CERT_KEY - path to private key for certificate to use for HTTPS endpoint
+  FLEET_SERVER_INSECURE_HTTP - expose Fleet Server over HTTP (not recommended; insecure)
+
+* Elastic Agent Fleet Enrollment
+  This enrolls the Elastic Agent into a Fleet Server. It is also possible to have this create a new enrollment token
+  for this specific Elastic Agent.
+
+  FLEET_ENROLL - set to 1 for enrollment to occur
+  FLEET_URL - URL of the Fleet Server to enroll into
+  FLEET_ENROLLMENT_TOKEN - token to use for enrollment
+  FLEET_TOKEN_NAME - token name to use for fetching token from Kibana
+  FLEET_TOKEN_POLICY_NAME - token policy name to use for fetching token from Kibana
+  FLEET_INSECURE - communicate with Fleet with either insecure HTTP or un-verified HTTPS
```

Comment:
This section mixes `Fleet` and `Fleet Server` - shouldn't the ENVs also be `FLEET_SERVER_xxx`?

Reply:
These are for the Agent itself to enroll. `FLEET_INSECURE` is different then `FLEET_SERVER_INSECURE_HTTP`.

Reply:
Should they be `KIBANA_FLEET_xyz` then instead? In the other ENVs you clearly separate between `FLEET_SERVER` and `KIBANA_FLEET`.

Reply:
These are not for Kibana, but are for the Agent itself. Just like the `FLEET_ENROLLMENT_TOKEN`. For the Agent its only `FLEET_*`.